### ISSUE TITLE
Failed to train Multi-Tasks of the same Modality

### ISSUE THREAD

**Issue by agemagician** [tensorflow/tensor2tensor#379]
Previously, there was a bug for training the different modailites  #338, but it did work with multi-tasks of the same modality.
Now after upgrading to latest commit, not even tasks from the same modality works !!!

Right now it only work for the same task only !!

Software:
Tensor2tensor: last commit 'a836d66'
Tensorflow: 1.3

Command:
```
t2t-trainer   --data_dir=~/t2t_data   --TMP_DIR=~/tmp/t2t_datagen   --problems=sentiment_imdb-translate_encs_wmt32k   --model=multi_model   --hparams_set=multi
model_tiny   --output_dir=~/t2t_train/multimodel_tiny_trans_sent
```

Error:
```
WARNING:tensorflow:RunConfig.uid (from tensorflow.contrib.learn.python.learn.estimators.run_config) is experimental and may change or be removed at any time, and without warning.
INFO:tensorflow:Creating experiment, storing model files in /home/ahmed/t2t_train/multimodel_tiny_trans_sent
INFO:tensorflow:datashard_devices: ['gpu:0']
INFO:tensorflow:caching_devices: None
INFO:tensorflow:Using config: {'_model_dir': '/home/ahmed/t2t_train/multimodel_tiny_trans_sent', '_session_config': gpu_options {
  per_process_gpu_memory_fraction: 0.95
}
allow_soft_placement: true
graph_options {
  optimizer_options {
  }
}
, '_tf_config': gpu_options {
  per_process_gpu_memory_fraction: 0.95
}
, '_tf_random_seed': None, '_save_checkpoints_secs': 600, '_master': '', '_save_checkpoints_steps': None, '_num_ps_replicas': 0, '_is_chief': True, '_environment': 'local', '_cluster_spec': <tensorflow.py
thon.training.server_lib.ClusterSpec object at 0x7f2647353278>, '_save_summary_steps': 100, '_keep_checkpoint_max': 20, '_log_step_count_steps': 100, '_task_type': None, '_task_id': 0, '_evaluation_master
': '', '_num_worker_replicas': 0, '_keep_checkpoint_every_n_hours': 10000}
WARNING:tensorflow:RunConfig.uid (from tensorflow.contrib.learn.python.learn.estimators.run_config) is experimental and may change or be removed at any time, and without warning.
WARNING:tensorflow:From /home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/monitors.py:269: BaseMonitor.__init__ (from tensorflow.contrib.learn.python.learn.moni
tors) is deprecated and will be removed after 2016-12-05.
Instructions for updating:
Monitors are deprecated. Please use tf.train.SessionRunHook.
INFO:tensorflow:batching_scheme = {'min_length': 0, 'window_size': 180, 'boundaries': [8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 22, 24, 26, 28, 30, 33, 36, 39, 42, 46, 50, 55, 60, 66, 72, 79, 86,
 94, 103, 113, 124, 136, 149, 163, 179, 196, 215, 236, 259, 284, 312, 343, 377, 414, 455, 500], 'max_length': 512, 'shuffle_queue_size': 540, 'batch_sizes': [60, 45, 45, 45, 36, 36, 36, 30, 30, 30, 20, 20
, 20, 20, 20, 18, 18, 15, 15, 12, 12, 12, 10, 10, 9, 6, 6, 6, 6, 5, 5, 4, 4, 4, 3, 3, 3, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
INFO:tensorflow:Reading data files from /home/ahmed/t2t_data/sentiment_imdb-train*
INFO:tensorflow:batching_scheme = {'min_length': 0, 'window_size': 180, 'boundaries': [8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 22, 24, 26, 28, 30, 33, 36, 39, 42, 46, 50, 55, 60, 66, 72, 79, 86,
 94, 103, 113, 124, 136, 149, 163, 179, 196, 215, 236, 259, 284, 312, 343, 377, 414, 455, 500], 'max_length': 512, 'shuffle_queue_size': 540, 'batch_sizes': [60, 45, 45, 45, 36, 36, 36, 30, 30, 30, 20, 20
, 20, 20, 20, 18, 18, 15, 15, 12, 12, 12, 10, 10, 9, 6, 6, 6, 6, 5, 5, 4, 4, 4, 3, 3, 3, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]}
INFO:tensorflow:Reading data files from /home/ahmed/t2t_data/translate_encs_wmt32k-train*
INFO:tensorflow:datashard_devices: ['gpu:0']
INFO:tensorflow:caching_devices: None
INFO:tensorflow:This model_fn took 4.027 sec.
Traceback (most recent call last):
  File "/home/ahmed/tensor2tensorp3/bin/t2t-trainer", line 96, in <module>
    tf.app.run()
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/platform/app.py", line 48, in run
    _sys.exit(main(_sys.argv[:1] + flags_passthrough))
  File "/home/ahmed/tensor2tensorp3/bin/t2t-trainer", line 92, in main
    schedule=FLAGS.schedule)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/trainer_utils.py", line 353, in run    hparams=hparams)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/learn_runner.py", line 209, in run
    return _execute_schedule(experiment, schedule)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/learn_runner.py", line 46, in _execute_schedule
    return task()
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/experiment.py", line 502, in train_and_evaluate
    self.train(delay_secs=0)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/experiment.py", line 280, in train
    hooks=self._train_monitors + extra_hooks)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/contrib/learn/python/learn/experiment.py", line 672, in _call_train
    hooks=hooks)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/estimator/estimator.py", line 241, in train
    loss = self._train_model(input_fn=input_fn, hooks=hooks)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/estimator/estimator.py", line 630, in _train_model
    model_fn_lib.ModeKeys.TRAIN)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/estimator/estimator.py", line 615, in _call_model_fn
    model_fn_results = self._model_fn(features=features, **kwargs)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/model_builder.py", line 348, in wrapping_model_fn
    return model_fn(model, features, mode, hparams, **kwargs)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/model_builder.py", line 163, in model_fn
    max_idx=len(hparams.problems) - 1)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/input_fn_builder.py", line 184, in cond_on_index
    lambda: cond_on_index(fn, index_tensor, max_idx, cur_idx + 1)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/util/deprecation.py", line 296, in new_func
    return func(*args, **kwargs)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/ops/control_flow_ops.py", line 1828, in cond
    orig_res_f, res_f = context_f.BuildCondBranch(false_fn)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/ops/control_flow_ops.py", line 1694, in BuildCondBranch
    original_result = fn()
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/input_fn_builder.py", line 184, in <lambda>
    lambda: cond_on_index(fn, index_tensor, max_idx, cur_idx + 1)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/input_fn_builder.py", line 179, in cond_on_index
    return fn(cur_idx)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/model_builder.py", line 131, in nth_model
    features, skip=(skipping_is_on and skip_this_one))
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/t2t_model.py", line 520, in model_fn
    transformed_features)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/models/multimodel.py", line 145, in model_fn_body_sharded
    hparams.hidden_size)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/expert_utils.py", line 222, in __call__
    outputs.append(fns[i](*my_args[i], **my_kwargs[i]))
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/models/slicenet.py", line 229, in embed_target_space
    target_space_id, 32, hidden_size, name="target_space_embedding")
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/layers/common_layers.py", line 196, in embedding
    embedding_var = tf.get_variable("kernel", [vocab_size, dense_size])
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/ops/variable_scope.py", line 1065, in get_variable
    use_resource=use_resource, custom_getter=custom_getter)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/ops/variable_scope.py", line 962, in get_variable
    use_resource=use_resource, custom_getter=custom_getter)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/ops/variable_scope.py", line 360, in get_variable
    validate_shape=validate_shape, use_resource=use_resource)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensor2tensor/utils/expert_utils.py", line 183, in daisy_chain_getter
    var = getter(name, *args, **kwargs)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/ops/variable_scope.py", line 352, in _true_getter
    use_resource=use_resource)
  File "/home/ahmed/tensor2tensorp3/lib/python3.5/site-packages/tensorflow/python/ops/variable_scope.py", line 682, in _get_single_variable
    "VarScope?" % name)
ValueError: Variable body/target_space_embedding/kernel does not exist, or was not created with tf.get_variable(). Did you mean to set reuse=None in VarScope?
```

**Comment by agemagician** [comment]
The problem still exists with the new release 1.2.6.

**Comment by agemagician** [comment]
@lukaszkaiser  Any idea when you are going to fix this issue along with issue number #338 ?
Without fixing these we cannot regenerate Google algorithm result or test it with other problems.

**Comment by agemagician** [comment]
The problems still exists with the new release 1.2.7
